### PR TITLE
PROD-592: Fix reveal status mixpanel event

### PR DESCRIPTION
**Pull request by neel-ds**
- Description
This update ensures accurate event recording by marking it as RevealTransactionPending instead of RevealTransactionSucceeded when the transaction status cannot be confirmed in some edge cases (e.g delay in sign txn)
 
- What are the steps to test that this code is working?
Try to reveal and sign txn after a few seconds of delay, you can verify from mixpanel dashboard in preview env

- Screen shots or recordings for UI changes
Screenshot 2024-12-19 at 6 17 59 PM

### CODE REVIEW

**Comment by neel-ds** [comment]
I mistakenly created branch from my previous work. Please ignore the other changes related to mysteryBox and migration.

File: scripts/addAllowlistUsers.ts [hunk 1]
```diff
@@ -31,8 +31,18 @@ async function main() {
   const csv = fs.readFileSync(csvFilePath, "utf8");
   const rows = csv.split(/\r?\n/);
 
-  // Filter out empty rows before processing
-  const validRows = rows.filter((address: string) => address.trim().length > 0);
+  // Filter out invalid addresses before adding to database
+  const validRows = rows.filter((address: string) => {
```
Comment (marvinmarnold) [inline scripts/addAllowlistUsers.ts:35]: These changes are unrelated to the PR

File: prisma/migrations/20241218180555_address_length_check/migration.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE "MysteryBoxAllowlist"
```
Comment (marvinmarnold) [inline prisma/migrations/20241218180555_address_length_check/migration.sql:1]: Unrelated to PR

File: app/hooks/useReveal.ts [hunk 3]
```diff
@@ -349,16 +350,31 @@ export function useReveal({
...
+      const revealResult = await getRevealPending(revealQuestionIds);
+
+      if (revealResult?.length !== 0) {
+        trackEvent(TRACKING_EVENTS.REVEAL_TRANSACTION_PENDING, {
+          transactionSignature: signature,
+          nftAddress: revealNft?.id,
+          nftType: revealNft?.type,
+          burnedAmount: reveal?.amount,
+          [TRACKING_METADATA.REVEAL_TYPE]:
+            revealQuestionIds.length > 1 ? REVEAL_TYPE.ALL : REVEAL_TYPE.SINGLE,
```
Comment (marvinmarnold) [inline app/hooks/useReveal.ts:362]: It is possible to reveal all with only one question. It's not a high priority to fix this. How hard would it be to differentiate based on the button the user is clicking instead of the number of questions?

File: app/hooks/useReveal.ts [hunk 4]
```diff
@@ -349,16 +350,31 @@ export function useReveal({
...
-        transactionSignature: signature,
-        nftAddress: revealNft?.id,
-        nftType: revealNft?.type,
-        burnedAmount: reveal?.amount,
-        [TRACKING_METADATA.REVEAL_TYPE]:
-          revealQuestionIds.length > 1 ? REVEAL_TYPE.ALL : REVEAL_TYPE.SINGLE,
-        [TRACKING_METADATA.QUESTION_ID]: reveal?.questionIds,
-        [TRACKING_METADATA.QUESTION_TEXT]: reveal?.questions,
-      });
+      const revealResult = await getRevealPending(revealQuestionIds);
```
Comment (marvinmarnold) [inline app/hooks/useReveal.ts:353]: revealResult not the best name. Maybe `pendingResults`?

File: app/actions/chompResult.ts [hunk 5]
```diff
@@ -251,6 +251,22 @@ export async function revealQuestions(
   revalidatePath("/application");
 }
 
+export async function getRevealPending(questionIds: number[]) {
```
Comment (marvinmarnold) [inline app/actions/chompResult.ts:254]: We should start putting all new actions into their own files. It's a preference, but it's a bit clearer to review the code that way.

Comment (marvinmarnold) [reply app/actions/chompResult.ts:254]: Also this name could be improved, maybe: getPendingChompResults

Comment (yash431garg) [reply app/actions/chompResult.ts:254]: We already have something for that https://github.com/gator-labs/chomp/blob/b9d8159639023e1149434985cbf5d1f14b0b3b93/app/actions/chompResult.ts#L341-L342

**Review by yash431garg — COMMENTED** [review]
Try asking devin to fix.

File: app/hooks/useReveal.ts (hunk 3 above)
Comment (neel-ds) [reply app/hooks/useReveal.ts:362]: Correct but our logic is bound to questions length so event will be recorded based on number of questions. Can you elaborate the intent?

File: app/actions/chompResult.ts (hunk 5 above)
Comment (neel-ds) [reply app/actions/chompResult.ts]: Thank you @yash431garg

File: app/hooks/useReveal.ts (hunk 3 above)
Comment (neel-ds) [reply app/hooks/useReveal.ts]: I updated all REVEAL mixpanel events with a flag passed as an args from button. REVEAL_TYPE will be based on button and not questionIds.

**Comment by neel-ds** [comment]
I updated all REVEAL mixpanel events with a flag passed as an args from button. REVEAL_TYPE will be based on button and not questionIds. We are GTG!
